src/dmd/optimize.d

```diff
@@ -487,7 +487,67 @@ Expression Expression_optimize(Expression e, int result, bool keepLvalue)
                         return true;
                     offset += v.offset;
                 }
-                else if (auto ve = e.isVarExp())
+            }
+
+            if (e.e1.isDotVarExp())
+            {
```

Comment:
This code block is also messed up (this condition _really_ begins on line 464 in master)